python/benchmarks/coco/train.py

```diff
@@ -0,0 +1,168 @@
+        return T.Compose(
+            [prepare_target, T.PILToTensor(), T.ConvertImageDtype(torch.float)]
+        )
+    elif benchmark == "train":
+        # https://github.com/pytorch/vision/blob/24890d718f5a73586ef093371912b5b37a5b0d46/references/detection/presets.py#L37
+        return T.Compose(
+            [
+                prepare_target,
+                T.RandomPhotometricDistort(),
+                T.RandomZoomOut(fill=list((123.0, 117.0, 104.0))),
+                T.RandomIoUCrop(),
+                T.RandomHorizontalFlip(),
+                T.PILToTensor(),
+                T.ConvertImageDtype(torch.float),
+            ]
+        )
+    else:
+        raise ValueError("Unsupported benchmark: ", benchmark)
+
+
+# https://github.com/pytorch/vision/blob/24890d718f5a73586ef093371912b5b37a5b0d46/references/detection/presets.py#L37
+transform = T.Compose(
+    [
+        prepare_target,
+        T.RandomPhotometricDistort(),
+        T.RandomZoomOut(fill=list((123.0, 117.0, 104.0))),
+        T.RandomIoUCrop(),
+        T.RandomHorizontalFlip(),
+        T.PILToTensor(),
+        T.ConvertImageDtype(torch.float),
+    ]
+)
+
+
+@click.command()
+@click.argument("uri")
+@click.option(
+    "-b", "--batch_size", default=32, help="set batch size", show_default=True
+)
+@click.option(
+    "-e",
+    "--epoch",
+    type=int,
+    default=10,
+    show_default=True,
+    help="set how many epoch to run",
+)
+@click.option(
+    "-w",
+    "--num_workers",
+    default=os.cpu_count(),
+    help="set Pytorch DataLoader workers",
+    show_default=True,
+)
+@click.option(
+    "-m",
+    "--model",
+    "model_name",
+    type=click.Choice(["ssd"]),
+    help="Choose the network to run",
+)
+@click.option(
+    "-B",
+    "--benchmark",
+    type=click.Choice(["io", "no_augmentation", "train"]),
+    default="train",
+    help="Specify the benchmark to run",
+    show_default=True,
+)
+@click.option(
+    "-f",
+    "--format",
+    "data_format",
+    type=click.Choice(["lance", "raw"]),
+    default="lance",
+    help="File format to train on",
+)
+def train(
+    uri: str,
+    batch_size: int,
+    epoch: int,
+    num_workers: int,
+    model_name: str,
+    data_format: str,
+    benchmark: str,
+):
+    if data_format == "lance":
+        dataset = lance.pytorch.data.LanceDataset(
+            uri,
+            columns=["image", "annotations.category_id", "annotations.bbox"],
+            batch_size=2,
+            transform=get_transform(benchmark),
+            # filter=(pc.field("split") == "train")
+        )
+        dp = IterableWrapper(dataset).shuffle(buffer_size=64)
+        train_loader = torch.utils.data.DataLoader(
+            dp,
+            num_workers=num_workers,
+            batch_size=batch_size,
+            collate_fn=collate_fn,
+            pin_memory=True,
+        )
+    elif data_format == "raw":
+        dataset = RawCocoDataset(uri, transform=get_transform(benchmark))
```

Comment:
do we not need the IterableWrapper for shuffle here?

Reply:
This is MapDataset, that DataLoader can shuffle it.